packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -105,50 +116,97 @@ export const searchBlockItems = (
 /**
  * Filters an item list given a search term.
  *
- * @param {Array} items       Item list
- * @param {string} searchTerm Search term.
- * @param {Object} config     Search Config.
- * @return {Array}            Filtered item list.
+ * @param {Array}  items       Item list
+ * @param {string} searchInput Search input.
+ * @param {Object} config      Search Config.
+ * @return {Array}             Filtered item list.
  */
-export const searchItems = ( items = [], searchTerm = '', config = {} ) => {
-	const normalizedSearchTerms = normalizeSearchTerm( searchTerm );
+export const searchItems = ( items = [], searchInput = '', config = {} ) => {
+	const normalizedSearchTerms = getNormalizedSearchTerms( searchInput );
 	if ( normalizedSearchTerms.length === 0 ) {
 		return items;
 	}
 
-	const defaultGetTitle = ( item ) => item.title;
-	const defaultGetKeywords = ( item ) => item.keywords || [];
-	const defaultGetCategory = ( item ) => item.category;
+	const rankedItems = items
+		.map( ( item ) => {
+			return [ item, getItemSearchRank( item, searchInput, config ) ];
+		} )
+		.filter( ( [ , rank ] ) => rank > 0 );
+
+	rankedItems.sort( ( [ , rank1 ], [ , rank2 ] ) => rank2 - rank1 );
+	return rankedItems.map( ( [ item ] ) => item );
+};
+
+/**
+ * Get the search rank for a given item and a specific search term.
+ * The better the match, the higher the rank.
+ * If the rank equals 0, it should be excluded from the results.
+ *
+ * @param {Object} item       Item to filter.
+ * @param {string} searchTerm Search term.
+ * @param {Object} config     Search Config.
+ * @return {number}           Search Rank.
+ */
+export function getItemSearchRank( item, searchTerm, config = {} ) {
+	const defaultGetName = ( it ) => it.name || '';
+	const defaultGetTitle = ( it ) => it.title;
+	const defaultGetKeywords = ( it ) => it.keywords || [];
+	const defaultGetCategory = ( it ) => it.category;
 	const defaultGetCollection = () => null;
 	const defaultGetVariations = () => [];
```

Comment:
It's a little tangential to the PR, but it _feels_ wasteful to reinstantiate these functions every time. But only by profiling could we decide whether to push them out into the module.

Reply:
It's minor to move them outside the function so i can do that now.